tests/test_query.py

```diff
@@ -377,3 +377,84 @@ async def test_trivial_unbatching(redis):
         },
         output["message"]
     )
+    
+@pytest.mark.asyncio
+@with_translator_overlay(
+    settings.kpregistry_url,
+    settings.normalizer_url,
+    kp_data={
+        "kp0":
+        """
+            MONDO:0008114(( category biolink:Disease ))
+            MONDO:0008114-- predicate biolink:related_to -->MESH:C035133
+            MESH:C035133(( category biolink:Gene ))
+        """,
+        "kp1":
+        """
+            MESH:C035133(( category biolink:Gene ))
+            MESH:C035133-- predicate biolink:related_to -->HP:0007430
+            HP:0007430(( category biolink:Protein ))
+        """,
+        "kp2":
+        """
+            HP:0007430(( category biolink:Protein ))
+            HP:0007430-- predicate biolink:related_to -->CHEBI:6801
+            CHEBI:6801(( category biolink:Disease ))
+            HGNC:6284(( category biolink:Gene ))
+            HP:0007430-- predicate biolink:related_to -->HGNC:6284
+        """
+    },
+    normalizer_data="""
+        MONDO:0008114 categories biolink:Disease
+        HP:0007430 categories biolink:Protein
+        MESH:C035133 categories biolink:Gene
+        HGNC:6284 categories biolink:Gene
+        CHEBI:6801 categories biolink:Disease
+        """
+)
+async def test_gene_protein_conflation(redis):
+    """Test conflation of biolink:Gene and biolink:Protein categories"""
```

Comment:
Please add documentation explaining why we have all of these edges and what each one is testing.